### DISCUSSION TITLE
integrate custom non-injected connector

### DISCUSSION THREAD

**Discussion by q20274982** [family/connectkit#214] [Q&A] [answered]
I have package a wallet that provides services through an SDK using the `Connector` class provided by wagmi.
I would like to know if this type of wallet can be integrated into connectkit through a PR.

**Comment by lochie** [comment, marked as answer]
If the wallet/connector is popular enough that we should support it then we will have a look at the PR, but currently we are mainly integrating support for [connectors that wagmi ships with](https://github.com/wagmi-dev/references/tree/main/packages/connectors#connectors), so if your connector is approved by wagmi we will most likely provide integrated support for it.
Worth noting that ConnectKit allows for custom wagmi connectors the same way as specified in [their documentation](https://wagmi.sh/react/client#connectors-optional) when creating a client.

**Reply by q20274982** [reply]
> Worth noting that ConnectKit allows for custom wagmi connectors the same way as specified in [their documentation](https://wagmi.sh/react/client#connectors-optional) when creating a client.

Thank you for your reply. Before this, I have tried using a custom wagmi connector as a solution. However, how can I display the logo of the custom wallet on the ConnectKit modal and link it to the ID of the custom wagmi connector?

**Reply by lochie** [reply]
There is an undocumented flow for this that will [soon be replaced with a better solution](https://github.com/family/connectkit/pull/184), but you can import [`supportedConnectors`](https://github.com/family/connectkit/blob/main/packages/connectkit/src/constants/supportedConnectors.tsx) from `connectkit` and push your connectors configuration object into this array and it should populate the UI with those assets.

**Reply by q20274982** [reply]
> There is an undocumented flow for this that will [soon be replaced with better solution](https://github.com/family/connectkit/pull/184)

I'm very excited about this. May I ask when it will be released approximately?
